### PR TITLE
Add codespell support (config, workflow to detect/not fix) and make it fix few typos

### PR DESCRIPTION
**Pull request by yarikoptic**
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

TODOs:
- [x] remove TEMP commit after verifying that we do or do not need provided custom action

### CODE REVIEW

**Comment by paraseba** [comment]
I like this! Let's make those checks pass and we can merge. Thank you @yarikoptic !

**Comment by yarikoptic** [comment]
oops -- my absent knowledge of rust -- do not even know how to add a comment correctly, will fix shortly.

But as above single failure showed, you have no run of pre-commit via CI or pre-commit.ci service, so I will keep github action for now.

File: icechunk/tests/test_s3_storage.rs [hunk 1]
```diff
@@ -75,7 +75,7 @@ pub async fn test_chunk_write_read() -> Result<(), Box<dyn std::error::Error>> {
     assert_eq!(Bytes::from_static(b"ello"), back);
 
     let back = storage.fetch_chunk(&id, &ByteRange::to_offset(3)).await?;
-    assert_eq!(Bytes::from_static(b"hel"), back);
+    assert_eq!(Bytes::from_static(b"hel"), back);  // codespell:ignore
```
Comment (paraseba) [inline icechunk/tests/test_s3_storage.rs:78]:
```suggestion
    assert_eq!(Bytes::from_static(b"hel"), back); // codespell:ignore
```

Comment (yarikoptic) [reply icechunk/tests/test_s3_storage.rs:78]: Thanks! I will take your help here ;)

**Comment by paraseba** [comment]
Thank you @yarikoptic !

**Review by paraseba — APPROVED** [review]

**Pull request merged**